### PR TITLE
9.0 base external dbsource

### PR DESCRIPTION
**Pull request by naousse**

### CODE REVIEW

**Comment by dreispt** [comment]
Needs a rebase on upstream 9.0

File: base_external_dbsource/README.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+* Click on Create to enter the following information:
+
+* Datasource name 
+* Pasword
+* Connector: Choose the database to which you want to connect
+* Connection string : Specify how to connect to database
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/128/9.0
```
Comment (max3903) [inline base_external_dbsource/README.rst]: @naousse `https://runbot.odoo-community.org/runbot/149/9.0` for server-tools

File: base_external_dbsource/models/base_external_dbsource.py [hunk 2]
```diff
@@ -62,58 +63,60 @@
...
+    - Microsoft SQL Server:
+      mssql+pymssql://username:%s@server:port/dbname?charset=utf8
+    - MySQL: mysql://user:%s@server:port/dbname
+    - ODBC: DRIVER={FreeTDS};SERVER=server.address;Database=mydb;UID=sa
+    - ORACLE: username/%s@//server.address:port/instance
+    - PostgreSQL:
+        dbname='template1' user='dbuser' host='localhost' port='5432' \
+        password=%s
+    - SQLite: sqlite:///test.db
+    """)
```
Comment (max3903) [inline base_external_dbsource/models/base_external_dbsource.py:84]: @naousse Can you add this `help` string as the placeholder as well?

Otherwise :+1:

**Comment by max3903** [comment]
@naousse Please add `\n` at the end of each line ot the placeholder

**Comment by naousse** [comment]
It doesn't work.
I've search without solution. Event with html entities. It seem's that odoo remove html entities before outputting the text.
Here is the only solution i found. 
https://www.odoo.com/fr_FR/forum/aide-1/question/how-to-insert-line-break-81846
If you find it useful, i will implement it

**Comment by max3903** [comment]
@naousse Let's put `Please check the tooltip for connection string examples.` in the placeholder

File: base_external_dbsource/__init__.py [hunk 3]
```diff
@@ -19,6 +19,4 @@
 #
 ##############################################################################
```
Comment (lasley) [inline base_external_dbsource/__init__.py:20]: Please switch to newer shortened header - https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: base_external_dbsource/demo/base_external_dbsource.xml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<openerp>
+<odoo>
     <data>
```
Comment (lasley) [inline base_external_dbsource/demo/base_external_dbsource.xml]: You can remove this

File: base_external_dbsource/views/base_external_dbsource.xml [hunk 5]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0"?>
+<odoo>
+    <data>
```
Comment (lasley) [inline base_external_dbsource/views/base_external_dbsource.xml]: You can remove this

File: base_external_dbsource/views/base_external_dbsource.xml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+                    <field name="conn_string"/>
+                </tree>
+            </field>
+        </record>
+
+        <record model="ir.ui.view" id="view_dbsource_form">
+            <field name="name">base.external.dbsource.form</field>
+            <field name="model">base.external.dbsource</field>
+            <field name="arch" type="xml">
+                <form string="External DB Source" version="7.0">
```
Comment (lasley) [inline base_external_dbsource/views/base_external_dbsource.xml]: Please remove version attr

**Comment by lasley** [comment]
Thanks for the submission @naousse - few comments inline. Additionally- would you mind increasing the test coverage a bit?

**Comment by max3903** [comment]
:+1:

**Comment by max3903** [comment]
@lasley Can you give us some time to review/merge here? Thank you.

I hope to see you in Belgium soon.

**Comment by lasley** [comment]
Will review today. Just got my ticket & lodging for Belgium - will be seeing you there @max3903 😄

**Comment by lasley** [comment]
Woot ❤️  @naousse thanks for cutting the test gap cut in half like that. LGTM 👍

**Comment by lasley** [comment]
@naousse - before I merge would you mind squashing please and thanks 😄

**Comment by max3903** [comment]
@lasley Github allows you to merge or to squash and merge at the same time ;)

**Pull request merged**